File: shared/src/commonMain/kotlin/com/fedetorresdev/iosmodularizationdemo/issues/IssuesListRepository.kt
package com.fedetorresdev.iosmodularizationdemo.issues

import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flow

interface IssuesListRepository {
    fun getAllIssues(): Flow<List<IssueModel>>
}

class IssuesListRepositoryImpl : IssuesListRepository {
    override fun getAllIssues(): Flow<List<IssueModel>> = flow {
        emit(issues)
    }
}

private val issues = listOf(
    IssueModel(
        id = "1",
        title = "Issue 1",
        description = "Description for issue 1",
        status = "Open"
    ),
    IssueModel(
        id = "2",
        title = "Issue 2",
        description = "Description for issue 2",
        status = "Closed"
    )
)

File: modularizediOSApp/Modules/Issues/Sources/Issues/Dependencies/IssuesListRepository.swift
//
//  IssuesListRepository.swift
//  Issues
//
//  Created by Federico Torres on 13/10/25.
//

import ComposableArchitecture
import CoreInterfaces
import Foundation

public struct IssuesListRepository: Sendable {
    public var getAllIssues: @Sendable () -> AsyncStream<[IssueUIModel]>
    
    public init(getAllIssues: @escaping @Sendable () -> AsyncStream<[IssueUIModel]>) {
        self.getAllIssues = getAllIssues
    }
}

extension IssuesListRepository: TestDependencyKey {
    public static var testValue: IssuesListRepository {
        return Self(
            getAllIssues: {
                return AsyncStream { continuation in
                    continuation.yield([
                        IssueUIModel(id: "TEST-1", title: "Test Issue 1", description: "Test description", status: "Open"),
                        IssueUIModel(id: "TEST-2", title: "Test Issue 2", description: "Test description", status: "Closed")
                    ])
                    continuation.finish()
                }
            }
        )
    }
}

public extension DependencyValues {
    var issuesListRepository: IssuesListRepository {
        get { self[IssuesListRepository.self] }
        set { self[IssuesListRepository.self] = newValue }
    }
}

